pkg/networkmanager/wireguard.jsx

```diff
@@ -0,0 +1,259 @@
+            setPrivateKey(result.wireguard["private-key"].v);
+        }
+
+        if (connection && connection[" priv"].path) {
+            getPrivateKey();
+        }
+    }, [model, connection]);
+
+    useEffect(() => {
+        if (privateKey === "") {
+            setPublicKey("");
+            return;
+        }
+
+        async function getPublicKey() {
+            try {
+                const key = await cockpit.spawn(["wg", "pubkey"], { err: 'message' }).input(privateKey.trim());
+                setPublicKey(key.trim());
+            } catch (e) {
+                console.error(e.message);
+                setPublicKey('');
+            }
+        }
+
+        getPublicKey();
+    }, [privateKey]);
+
+    function onSubmit() {
+        // Validate Addresses before submit
+        let addr;
+        try {
+            addr = stringToAddresses(addresses);
+        } catch (e) {
+            setDialogError(e.message);
+            return;
+        }
+
+        function createSettingsObj() {
+            return {
+                ...settings,
+                connection: {
+                    ...settings.connection,
+                    id: `con-${iface}`,
+                    interface_name: iface,
+                    type: 'wireguard'
+                },
+                wireguard: {
+                    private_key: privateKey.trim(),
+                    listen_port: +listenPort,
+                    peers: peers.map(peer => ({ ...peer, allowedIps: peer.allowedIps.split(',') })),
+                },
+                ipv4: {
+                    addresses: addr,
+                    method: 'manual',
+                    dns: [],
+                    dns_search: []
+                }
+            };
+        }
+
+        dialogSave({
+            connection,
+            dev,
+            model,
+            settings: createSettingsObj(),
+            onClose: Dialogs.close,
+            setDialogError
+        });
+    }
+
+    return (
+        <NetworkModal title={_("WireGuard settings")} onSubmit={onSubmit} dialogError={dialogError} idPrefix={idPrefix} submitDisabled={!iface || !addresses || !privateKey}>
+            <Name idPrefix={idPrefix} iface={iface} setIface={setIface} />
+            <FormGroup label={_("Private key")} fieldId={idPrefix + '-private-key-input'}>
+                <InputGroup>
+                    <TextInput id={idPrefix + '-private-key-input'} className='pf-v5-u-font-family-monospace' value={privateKey} onChange={(_, val) => setPrivateKey(val)} />
+                    <Button variant='control'
+                        onClick={async () => {
+                            try {
+                                const key = await cockpit.spawn(["wg", "genkey"]);
+                                setPrivateKey(key.trim());
+                            } catch (e) {
+                                setDialogError(e.message);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19024-20230721-095543-753c16c5-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)